### ISSUE TITLE
33 warnings on #516 macro implementation

### ISSUE THREAD

**Issue by seratch** [scalatra/scalatra#520]
Although I should have checked in detail earlier, #516 macro implementation brought 33 warnings to core project.

@dozed Do you have time to fix these warnings until 2.4.0 release? I'm planning to release next RC (hopefully the last RC version) within one or two weeks.

```

[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:12: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def rescopeExpression[C <: Context](c: C)(expr: c.Expr[Any]): c.Expr[Any] = {
[warn]                              ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:19: class AsyncResult in package scalatra is deprecated: AsyncResult is no longer required and will be removed in a coming release.
[warn]     if (expr.actualType <:< implicitly[TypeTag[AsyncResult]].tpe) {
[warn]                                        ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:29: method newTypeName in trait Names is deprecated: Use TypeName instead
[warn]       val clsName = newTypeName(c.fresh("cls"))
[warn]                     ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:29: method fresh in trait Names is deprecated: Use freshName instead
[warn]       val clsName = newTypeName(c.fresh("cls"))
[warn]                                   ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:30: method newTermName in trait Names is deprecated: Use TermName instead
[warn]       val resName = newTermName(c.fresh("res"))
[warn]                     ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:30: method fresh in trait Names is deprecated: Use freshName instead
[warn]       val resName = newTermName(c.fresh("res"))
[warn]                                   ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:35: method newTermName in trait Names is deprecated: Use TermName instead
[warn]             case q"$a.this.request" => Select(This(clsName), newTermName("request"))
[warn]                                                              ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:36: method newTermName in trait Names is deprecated: Use TermName instead
[warn]             case q"$a.this.response" => Select(This(clsName), newTermName("response"))
[warn]                                                               ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:43: method resetLocalAttrs in trait Typers is deprecated: Use `c.untypecheck` instead
[warn]       val untypedExpr = c.resetLocalAttrs(expr.tree.duplicate)
[warn]                           ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:67: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def addRouteGen[C <: Context](c: C)(method: c.Expr[HttpMethod], transformers: Seq[c.Expr[RouteTransformer]], action: c.Expr[Any]): c.Expr[Route] = {
[warn]                        ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:74: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def beforeImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Unit] = {
[warn]                     ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:81: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def afterImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Unit] = {
[warn]                    ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:88: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def getImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                  ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:92: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def postImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                   ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:96: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def putImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                  ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:100: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def deleteImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                     ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:104: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def optionsImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                      ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:108: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def headImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                   ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:112: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def patchImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(action: c.Expr[Any]): c.Expr[Route] = {
[warn]                    ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:116: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def trapImpl(c: Context)(codes: c.Expr[Range])(block: c.Expr[Any]): c.Expr[Unit] = {
[warn]                   ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:123: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def trapCodeImpl(c: Context)(code: c.Expr[Int])(block: c.Expr[Any]): c.Expr[Unit] = {
[warn]                       ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:130: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def notFoundImpl(c: Context)(block: c.Expr[Any]): c.Expr[Unit] = {
[warn]                       ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:142: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def methodNotAllowedImpl(c: Context)(block: c.Expr[Set[HttpMethod] => Any]): c.Expr[Unit] = {
[warn]                               ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:155: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def errorImpl(c: Context)(handler: c.Expr[ErrorHandler]): c.Expr[Unit] = {
[warn]                    ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:182: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def makeAsynchronously[C <: Context](c: C)(block: c.Expr[Any]): c.Expr[Any] = {
[warn]                               ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:184: method resetLocalAttrs in trait Typers is deprecated: Use `c.untypecheck` instead
[warn]     val block1 = c.resetLocalAttrs(block.tree.duplicate)
[warn]                    ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:185: method typeCheck in trait Typers is deprecated: Use `c.typecheck` instead
[warn]     c.Expr[Any](c.typeCheck(q"""asynchronously($block1)()"""))
[warn]                   ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:188: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def asyncGetImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Route] = {
[warn]                       ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:193: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def asyncPostImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Route] = {
[warn]                        ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:198: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def asyncPutImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Route] = {
[warn]                       ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:203: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def asyncDeleteImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Route] = {
[warn]                          ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:208: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def asyncOptionsImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Route] = {
[warn]                           ^
[warn] /scalatra/core/src/main/scala/org/scalatra/CoreDslMacros.scala:213: type Context in package macros is deprecated: Use blackbox.Context or whitebox.Context instead
[warn]   def asyncPatchImpl(c: Context)(transformers: c.Expr[RouteTransformer]*)(block: c.Expr[Any]): c.Expr[Route] = {
[warn]                         ^
[warn] 33 warnings found
```

**Comment by seratch** [comment]
Since the deprecations are relatively simple ones, I just fixed and made a pull request at #521. After successful Travis CI build, I'll merge the PR and release 2.4.0.RC2-1.

**Comment by seratch** [comment]
I've understood fixing these warnings means dropping Scala 2.10 right now. That's overquick.

**Comment by dozed** [comment]
@seratch I created a patch here: https://github.com/scalatra/scalatra/pull/522

**Comment by seratch** [comment]
Fixed by #522
